pkgs/tools/package-management/nix/default.nix

```diff
@@ -46,67 +46,29 @@ let
     sha256 = "sha256-tI5nKU7SZgsJrxiskJ5nHZyfrWf5aZyKYExM0792N80=";
   };
 
-in lib.makeExtensible (self: {
-  nix_2_3 = (common rec {
-    version = "2.3.16";
-    src = fetchurl {
-      url = "https://nixos.org/releases/nix/nix-${version}/nix-${version}.tar.xz";
-      sha256 = "sha256-fuaBtp8FtSVJLSAsO+3Nne4ZYLuBj2JpD2xEk7fCqrw=";
+  inherit (builtins) baseNameOf dirOf listToAttrs map;
+  inherit (lib.lists) flatten;
+  majorVersions = [ ./2 ];
+  getSubDirs =
+    d: lib.mapAttrsToList
+      (k: v: lib.optional (v=="directory") (d + "/${k}"))
+      (builtins.readDir d);
+  versionDirs =
+    flatten (map getSubDirs majorVersions);
+  dirToKeyVal = d:
+    let
+      minor = baseNameOf d;
+      major = baseNameOf (dirOf d);
+    in {
+      name = "nix_${major}_${minor}";
+      value = callPackage (import d) {
+        inherit common patch-sqlite-exception boehmgc-nix_2_3;
+      };
     };
-  }).override { boehmgc = boehmgc-nix_2_3; };
-
-  nix_2_4 = throw "nixVersions.nix_2_4 has been removed";
-
-  nix_2_5 = throw "nixVersions.nix_2_5 has been removed";
-
-  nix_2_6 = throw "nixVersions.nix_2_6 has been removed";
-
-  nix_2_7 = throw "nixVersions.nix_2_7 has been removed";
-
-  nix_2_8 = throw "nixVersions.nix_2_8 has been removed";
-
-  nix_2_9 = throw "nixVersions.nix_2_9 has been removed";
```

Comment:
IMO those throws should stay in this file.

Reply:
Unfortunately if they remain in this file, the goal of avoiding merge conflicts cannot be achieved (unless you see a way to do so -- if so please make a concrete suggestion).

Reply:
Alright, see 63a6de83ca1149fdc4a9d8930a311c7874315c2a which is as close as I could get.